nixos/modules/services/hardware/dockd.nix

```diff
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ...}:
+with lib;
+let
+  cfg = config.services.dockd;
+in
+{
+  ###############################################
+  options.services.dockd = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        If enabled, run dockd as a systemd user service.
+      '';
+    };
+
+    dockedConfPath = mkOption {
```

Comment:
I appreciate the effort you took to make this configurable, but I don't think it's necessary to maintain this solution. Most services simply define [a configuration in etc](https://nixos.org/nixos/options.html#environment.etc).

Of course your solution is more powerful, because it allows the possibility to have a mutable config file, but is that worth the overhead of a different solution?

I may well be wrong, because I don't know anything about operating `dockd`.

Reply:
Thank you for your comment.

About the configs... dockd  has two config files

1. The `docked.conf` this file has to be _writable_ to be set with `dockd --config docked` (executed by the user running under the graphical session), this command captures the current monitor/screens configuration in `xrandr` and create an ini file with those values. (on a predefined location *hard coded* in the executable)

2. The `undocked.conf` this file has to be _writable_ to be set with `dockd --config undocked` (executed by the user running under the graphical session), this command captures the current monitor/screens configuration in `xrandr` and create an ini file with those values (on a predefined location *hard coded* in the executable). Basically this is the same as (1) but the result will be written into another path.

And two _hooks_/_executables_:

1. The `dock.hook` script/executable, it will be executed when the laptop gets removed from the dock, you can put any arbitrary command in this file.
 
 2. The `undock.hook` script/executable, it will be executed when the laptop gets removed from the dock, you can put any arbitrary command in this file.

I'm not familiar with the options to manage an etc file.. but that seems like a potential good solution, the only problem is that, if you want to change the configuration (`dockd --config docked|undocked`) the user would need to copy those new values into the `configuration.nix` otherwise the changes will be overridden on the next `nixos-rebuild`.

Reply:
So it's more of configuration management tool. That makes sense. Could you add this information in the option descriptions?